### ISSUE TITLE
services: make the AWS SDK configurations more obvious and easier to work with

### ISSUE THREAD

**Issue by JadenSimon** [aws/aws-toolkit-vscode#2726]
## Problem
The SDKs pull in configuration data from a lot of places: environment variables, shared INI files, client constructors (within the Toolkit), etc.

Developers should be able to easily figure out what state the SDK/Toolkit is in when making API calls. An environment variable like `AWS_USE_DUALSTACK_ENDPOINT` can cause the Toolkit to fail in mysterious ways with little explanation as to _why_ things are failing. Also, exposing more of the internal configurations could eventually allow users to do things like define their own endpoints.

**Comment by justinmk3** [comment]
This might fit in the "toolkit dev dashboard" concept.

**Comment by JadenSimon** [comment]
> This might fit in the "toolkit dev dashboard" concept.

Yup definitely could. Features relating to the AWS SDKs become pretty interesting as I'm sure many of our users are also using the SDKs.
